### DISCUSSION TITLE
Live reload / watch mode with Nodemon

### DISCUSSION THREAD

**Discussion by michaelbromley** [vendurehq/vendure#2036] [How-to Guides]
If you'd like your dev server to automatically restart whenever a file is changed, you can use [nodemon](https://www.npmjs.com/package/nodemon).

1. Install nodemon
   ```
   npm install -D nodemon
   ```
2. Create a config file for nodemon:
   ```json
   {
     "watch": ["src/",".env"],
     "ext": "ts,js,json",
     "ignore": ["src/plugins/**/ui/*", "src/migrations"],
     "exec": "npm run concurrently"
   }
   ```
3. Update your `dev` script in package.json:
   ```
   "dev:" "nodemon -x concurrently npm:dev:*",
   ```

Now, when you run `npm run dev`, any changes to your source code will automatically restart your dev server.

**Comment by chregu** [comment]
If you do `TS_NODE_TRANSPILE_ONLY=true`before, the reload will also be faster (you just won't have TypeScript checks, but maybe your IDE does that already

**Comment by chungchi300** [comment]
I am new to vendure,  but   "ignore": ["src/plugins/**/ui/*", "src/migrations", "src/gql"], is needed to modified or else it reload forever , I guess vendure generate src/gql every time
